src/algebra/linear_ordered_comm_group_with_zero.lean

```diff
@@ -52,6 +52,112 @@ instance [linear_ordered_add_comm_group_with_top α] :
   ..multiplicative.linear_ordered_comm_monoid_with_zero,
   ..multiplicative.nontrivial }
 
+section monoid
+
+variables [monoid α] [preorder α]
+
+section left
+variable [covariant_class α α (*) (≤)]
+lemma left.pow_le_one_of_le : ∀ {n : ℕ} {x : α}, x ≤ 1 → x^n ≤ 1
+| 0       x _ := (pow_zero x).le
+| (n + 1) x H := calc x ^ n.succ = x * x ^ n : pow_succ x n
+                             ... ≤ x * 1     : mul_le_mul_left' (left.pow_le_one_of_le H) x
+                             ... = x         : mul_one x
+                             ... ≤ 1         : H
+
+alias left.pow_le_one_of_le ← pow_le_one_of_le_one
+
+lemma left.one_le_pow_of_le : ∀ {n : ℕ} {x : α}, 1 ≤ x → 1 ≤ x^n
+| 0       x _ := (pow_zero x).symm.le
+| (n + 1) x H := calc 1 ≤ x          : H
+                    ... = x * 1      : (mul_one x).symm
+                    ... ≤ x * x ^ n  : mul_le_mul_left' (left.one_le_pow_of_le H) x
+                    ... = x ^ n.succ : (pow_succ x n).symm
+
+alias left.one_le_pow_of_le ← one_le_pow_of_one_le'
+
+lemma eq_one_of_pow_eq_one {α : Type*} [monoid α] [linear_order α]
+  [covariant_class α α (*) (≤)]
+  [covariant_class α α (function.swap (*)) (≤)]
+  {x : α} {n : ℕ} (hn : n ≠ 0) (H : x ^ n = 1) : x = 1 :=
```

Comment:
Move out of the `left` section. Maybe maybe a `both` section together with `pow_le_pow_of_le`?

Can you prove `iff` versions of the previous two lemmas (under `linear_order` and both covariant assumptions)? Then this proof is a ~1 liner.

Also, can you add the `iff` version of this?

Reply:
In the meanwhile, this lemma appeared in mathlib: it is called `pow_eq_one_iff` and it is in `src/algebra/group_power/order`.  I simply removed it from this PR.